File: public-api/src/main/java/com/shopleech/publicapi/bll/service/model/ICategoryService.java
package com.shopleech.publicapi.bll.service.model;

import com.shopleech.publicapi.domain.Category;

import java.util.List;

/**
 * @author Ahto Jalak
 * @since 05.04.2023
 */
public interface ICategoryService {
    Category add(Category data);

    Category get(int id) throws Exception;

    List<Category> getAll();

    Category update(Integer id, Category category);

    Integer remove(Integer id);

    List<Category> getAllByCategoryId(Integer id);

    List<Category> getAllByCategory(Category entity);
}

File: price-comparison-client1/src/bll/service/model/ICategoryService.ts
import type { ICategory } from '@/dal/domain/ICategory'
import type { IServiceResult } from '@/bll/service/model/IServiceResult'

/**
 * @author Ahto Jalak
 * @since 12.04.2023
 */
export interface ICategoryService {
    getAllByCategoryId (categoryId: number | null): Promise<IServiceResult<ICategory[]>>

    findByName (searchName: string): Promise<IServiceResult<ICategory[]>>

    add (category: ICategory): Promise<IServiceResult<ICategory>>

    getAll (): Promise<IServiceResult<ICategory[]>>
}
